### PR TITLE
Finish migration to argument-less with() method.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

**Comment by JakeWharton** [comment]
cc @SalvatoreT

**Comment by SalvatoreT** [comment]
Does the name "`with`" still make sense now that there's no context directly passed-in?

**Comment by JakeWharton** [comment]
No. Probably should be `get()`, but it'd also be great to figure out a way to just make `load()`, etc. static methods so you don't even need `get()`.

**Comment by JakeWharton** [comment]
@dnkoutso can you review? weigh in?

File: picasso/src/main/java/com/squareup/picasso/Picasso.java [hunk 1]
```diff
@@ -673,31 +673,21 @@ void shutdown() {
...
-    if (context == null) {
-      throw new IllegalArgumentException("context == null");
-    }
+  public static Picasso with() {
     if (singleton == null) {
       synchronized (Picasso.class) {
         if (singleton == null) {
-          singleton = new Builder(context).build();
+          if (PicassoProvider.context == null) {
+            throw new IllegalStateException("context == null");
```
Comment (NightlyNexus) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: Drive-by thought:
This would only happen if you remove the ContentProvider in your manifest, right?
Should the error message suggest checking for that?
